app/src/main/java/net/squanchy/service/firebase/FirebaseDbService.java

```diff
@@ -85,17 +86,22 @@ public FirebaseDbService(DatabaseReference database) {
     }
 
     public Observable<FirebaseFavorites> favorites(String userId) {
-        String path = String.format(Locale.US, FAVORITES_NODE, userId);
-
-        return observeOptionalChild(path, FirebaseFavorites.class)
-                .map(optionalFavorites -> optionalFavorites.or(FirebaseFavorites.empty()));
+        return userData(userId)
+                .map(firebaseUserData -> Optional.fromNullable(firebaseUserData.favorites))
+                .map(FirebaseFavorites::new);
     }
 
     public Observable<FirebaseAchievements> achievements(String userId) {
-        String path = String.format(Locale.US, ACHIEVEMENTS_NODE, userId);
+        return userData(userId)
+                .map(firebaseUserData -> Optional.fromNullable(firebaseUserData.achievements))
```

Comment:
this is gonna crash if there's a null, RxJava 2 doesn't support propagating null values. But then I don't see how `userData` would return a null, so maybe this should be wrapped in an `Optional`.